pkg/controllers/cloud/networkpolicy.go

```diff
@@ -872,47 +745,204 @@ func (a *appliedToSecurityGroup) updateRules(r *NetworkPolicyReconciler) error {
 		return fmt.Errorf("unable to get networkPolicy with key %s from indexer: %w", a.id.Name, err)
 	}
 	if len(nps) == 0 {
-		if a.hasMembers {
-			r.Log.V(1).Info("AppliedToSecurityGroup clear members with no rules", "Name", a.id.Name)
-			ch := securitygroup.CloudSecurityGroup.UpdateSecurityGroupMembers(&a.id, nil, false)
-			go func() {
-				err := <-ch
-				r.cloudResponse <- &securityGroupStatus{sg: a, op: securityGroupOperationClearMembers, err: err}
-			}()
-			return nil
-		}
-		// No need to update appliedToSecurityGroup with no members.
-		r.Log.V(1).Info("AppliedToSecurityGroup clear rules", "Name", a.id.Name)
-		a.hasRules = false
+		a.clearMember(r)
 		return nil
 	}
-	irules, erules := a.computeRules(nps)
-	if irules == nil || erules == nil {
-		return nil
+
+	// sort by np creation time in ascending order.
+	sort.Slice(nps, func(i, j int) bool {
+		return nps[i].(*networkPolicy).GetCreationTimestamp().Unix() < nps[j].(*networkPolicy).GetCreationTimestamp().Unix()
+	})
+	for _, obj := range nps {
+		np := obj.(*networkPolicy)
+		a.updateANPRules(r, np)
+	}
+
+	return nil
+}
+
+// updateANPRules invokes cloud plug-in to update rules of appliedToSecurityGroup for a single ANP.
+func (a *appliedToSecurityGroup) updateANPRules(r *NetworkPolicyReconciler, np *networkPolicy) {
+	if !a.isReady() || a.deletePending || !np.rulesReady {
+		return
+	}
+
+	nps, err := r.networkPolicyIndexer.ByIndex(networkPolicyIndexerByAppliedToGrp, a.id.Name)
+	if err != nil {
+		r.Log.Error(err, "get networkPolicy indexer", "sg", a.id.String())
+		r.sendRuleRealizationStatus(&np.NetworkPolicy, err)
+		return
+	}
+	if len(nps) == 0 {
+		a.clearMember(r)
+		return
+	}
+	_, targetRules := a.combineRules(nps)
```

Comment:
nit: Add a one line comment to explain why we are calling a.combineRules(nps) and a.combineRules([]interface{}{np})